redisgraph/query_result.py

```diff
@@ -127,9 +128,9 @@ def parse_scalar(self, cell):
             scalar = int(value)
 
         elif scalar_type == ResultSetScalarTypes.PROPERTY_BOOLEAN:
-            if value == "true":
+            if value == b"true":
```

Comment:
Are we guarantee to always be dealing with byte strings?

Isn't it possible for `value` to be "true" when the redis client (redis-py) is created with:`decode_responses=True`
`redis.Redis(host='localhost', port=6379, decode_responses=True)` ?

Reply:
If we run the instance with `decode_responses=False`, then we are just checking `b'true' == "true"` which returns false and the value is `None`. We actually don't know that `decode_responses` will always be true.

Reply:
I think we can better fix this by doing:
```suggestion
            raw_value = cell[1]
            value = raw_value.decode() if isinstance(raw_value, bytes) else raw_value
```

Reply:
Sorry for dragging this, but wouldn't it be possible for a user to actually store binary data within the node / edge attributes, in which case, is it OK to decode the value ?

Reply:
I get your point. But what if an instance is ran with `decode_response=False`, this is the response we will be getting

```
query = 'MATCH (a:Employee)-[t]->(b) RETURN a, exists(a)'

Unknown boolean type
Unknown boolean type
Unknown boolean type

+------------------------+--------------+
|          b'a'          | b'exists(a)' |
+------------------------+--------------+
| {b'name':"b'Gnanesh'"} |     None     |
|  {b'name':"b'Bill'"}   |     None     |
|  {b'name':"b'Satya'"}  |     None     |
+------------------------+--------------+

internal execution time 0.267071

```

Reply:
I think when you run redis with `decode_responses=True`, you automatically decode binary data.
```
>>> import redis
>>> r = redis.Redis()
>>> x = b'hello'
>>> x
b'hello'
>>> r.set('name', x)
True
>>> r.get('name')
b'hello'
>>> r = redis.Redis(decode_responses=True)
>>> r.get('name')
'hello'
>>> 

```

Reply:
Agree, when using `redis.Redis(decode_responses=True)` redis-py automatically decodes strings,
my concern is to always decode value as you've suggested, even when decode_responses=False.

I think a safer approach would be to decode value only within the condition
` elif scalar_type == ResultSetScalarTypes.PROPERTY_BOOLEAN:
        value = value.decode() if isinstance(value, bytes) else value
            if value == "true":
                scalar = True
            elif value == "false":
                scalar = False
            else:
                print("Unknown boolean type\n")`

Thoughts?

Reply:
Great. I think we have to do the same with `ResultSetScalarTypes.PROPERTY_STRING` as well. Cause `str(b'hello')` becomes `"b'hello'"`. I'll make a commit if it's okay.

Reply:
As currently we don't have a "binary" type within RedisGraph I think String will serve as both binary and none binary data, as such, those who do wish to work with decoded strings should init their redis-py connection with `decode_responses=True`

For the moment, let's introduce the change only to boolean type.

Reply:
Okay. Added change.